operator/src/monitoring.rs

```diff
@@ -0,0 +1,591 @@
+        &self,
+        task_id: &str,
+        revision: &ReportingTaskRevision,
+    ) -> String {
+        format!(
+            "http://{}/nifi-api/reporting-tasks/{}?version={}&clientId={}&disconnectedNodeAcknowledged=false",
+            self.http_host(),
+            task_id, revision.version, revision.client_id.as_deref().unwrap_or("1")
+        )
+    }
+
+    /// Url to change the status of an existing task.
+    pub fn update_reporting_task_status_url(&self, task_id: &str) -> String {
+        format!(
+            "http://{}/nifi-api/reporting-tasks/{}/run-status",
+            self.http_host(),
+            task_id
+        )
+    }
+}
+
+pub struct NifiRestClient {
+    pub client: reqwest::Client,
+    pub headers: HeaderMap,
+}
+
+impl NifiRestClient {
+    pub fn new(client: reqwest::Client) -> Self {
+        let headers: Vec<(&'static str, &str)> = vec![
+            ("Accept-Encoding", "gzip, deflate, br"),
+            ("Content-Type", "application/json"),
+            ("Accept", "application/json"),
+        ];
+
+        let mut header_map = HeaderMap::new();
+
+        for (key, value) in headers {
+            match HeaderValue::from_str(value) {
+                Ok(val) => {
+                    header_map.insert(key, val);
+                }
+                Err(err) => {
+                    debug!(
+                        "Invalid header item [{} -> {}] for monitoring requests: {}",
+                        key,
+                        value,
+                        err.to_string()
+                    );
+                }
+            }
+        }
+
+        NifiRestClient {
+            client,
+            headers: header_map,
+        }
+    }
+
+    /// Find the "StackablePrometheusReportingTask" in the NiFi cluster.
+    ///
+    /// # Arguments
+    /// * `monitoring_info` - Pod monitoring info including container ports, uid etc.
+    /// * `nifi_version` - The NiFi version currently used by the operator.
+    ///
+    pub async fn find_reporting_task(
+        &self,
+        monitoring_info: &[PodMonitoringInfo],
+        nifi_version: &str,
+    ) -> Result<Option<ReportingTask>, NifiMonitoringError> {
+        let tasks = self.list_reporting_tasks(monitoring_info).await?;
+        for task in tasks {
+            if let Some(ReportingTaskComponent {
+                typ,
+                name,
+                bundle:
+                    ReportingTaskComponentBundle {
+                        group,
+                        artifact,
+                        version,
+                    },
+                ..
+            }) = &task.component
+            {
+                if typ != PROMETHEUS_REPORTING_TASK_TYPE && name != &build_task_name() {
+                    continue;
+                }
+
+                if group != PROMETHEUS_REPORTING_TASK_BUNDLE_GROUP
+                    && artifact != PROMETHEUS_REPORTING_TASK_BUNDLE_ARTIFACT
+                    && version != nifi_version
```

Comment:
I think this should be || instead of &&.

Reply:
Good point thanks. Will adapt.